processing/data_collection/gazette/spiders/pr_curitiba.py

```diff
@@ -0,0 +1,119 @@
+        """
+        todays_date = dt.date.today()
+        current_year = todays_date.year
+        for year in range(current_year, 2014, -1):
+            yield self.scrap_year(response, year)
+
+    def scrap_year(self, response, year):
+        return scrapy.FormRequest.from_response(
+            response,
+            formdata={
+                'ctl00$cphMasterPrincipal$ddlGrAno': str(year)
+            },
+            callback=self.parse_year
+        )
+
+
+    def parse_year(self, response):
+        for i in range(12):
+            yield self.scrap_month(response, i)
+
+    def scrap_month(self, response, month):
+        return scrapy.FormRequest.from_response(
+            response,
+            formdata={
+                '__EVENTTARGET': 'ctl00$cphMasterPrincipal$TabContainer1',
+                '__EVENTARGUMENT': 'activeTabChanged:{}'.format(month),
+                'ctl00_cphMasterPrincipal_TabContalegacyDealPooliner1_ClientState' : '{{"ActiveTabIndex":{},"TabState":[true,true,true,true,true,true,true,true,true,true,true,true]}}'.format(month)
+            },
+            meta={ "month" : month },
+            callback=self.parse_month
+        )
+
+    def parse_month(self, response):
+        page_count = len(response.css(".grid_Pager:nth-child(1) table td").extract())
+        month = response.meta["month"]
+        #The first page of pagination cannot be accessed by page number
+        yield scrapy.FormRequest.from_response(
+            response,
+            formdata={
+                '__EVENTTARGET': 'ctl00$cphMasterPrincipal$TabContainer1',
+                'ctl00_cphMasterPrincipal_TabContalegacyDealPooliner1_ClientState' : '{{"ActiveTabIndex":{},"TabState":[true,true,true,true,true,true,true,true,true,true,true,true]}}'.format(month),
+                '__EVENTARGUMENT': 'activeTabChanged:{}'.format(month),
+            },
+            callback=self.parse_page,
+        )
+        for page_number in range(2, page_count + 1):
+            yield scrapy.FormRequest.from_response(
+                response,
+                formdata={
+                    '__EVENTARGUMENT' : 'Page${}'.format(page_number),
+                    '__EVENTTARGET' : 'ctl00$cphMasterPrincipal$gdvGrid2'
+                },
+                callback=self.parse_page,
+            )
+    def parse_page(self, response):
+        row = response.css(".grid_Row")
+        numbers = row.css("td:nth-child(1) span ::text").extract()
+        pdf_dates = row.css("td:nth-child(2) span ::text").extract()
+        ids = row.css("td:nth-child(3) a ::attr(data-teste)").extract()
+        for i in range(len(numbers)):
+            number = numbers[i]
+            pdf_date = pdf_dates[i]
+            id = ids[i]
+            parsed_date = parse(f'{pdf_date}', languages=['pt']).date()
+            if id == '0':
+                yield scrapy.FormRequest.from_response(
+                    response,
+                    headers = {'user-agent': 'Mozilla/5.0'},
```

Comment:
Is it really necessary to change the `User-Agent` in this request? You are doing similar requests in other places and didn't changed it. If this is really required for this website, I suggest to set a custom user-agent for all the requests made by this spider.

Include a `custom_settings` in the spider as you can see here https://doc.scrapy.org/en/latest/topics/settings.html#settings-per-spider and set the desired `USER_AGENT` key (https://doc.scrapy.org/en/latest/topics/settings.html#user-agent)